frontend/src/metabase/css/components/buttons.css

```diff
@@ -22,7 +22,9 @@
   border-color: color-mod(var(--color-border) blackness(12%));
   background: var(--color-bg-light);
   transition: all 200ms linear;
-  transition-property: color, border-color, background-color;
+  /* XXX: `visibility` and `opacity` are needed otherwise it won't transition when users click on the apply button, since it's essentially the hover state */
+  transition-property: color, border-color, background-color, visibility,
```

Comment:
`visibility` and `opacity` are needed otherwise it won't transition and would disappear immediately when users click on the apply button, since it's essentially the hover state

Reply:
If we need `visibility` transition for the `Apply` button, I believe we should add it in a styled component for the apply button and not in global css. Thoughts?

Reply:
I could, but don't you think we should make all buttons support fade in and out? It feels more something that should be changed in the core component than a one-off thing.

This feels similar to a lot of changes I've seen done in visualization settings where we ended up changing the style of the component than adding styles to the `.styled.*` file.

What do you think?